reposilite-frontend/src/store/console/connection.js

```diff
@@ -67,37 +73,44 @@ export default function useConsole() {
     command.value = history.value[historyIdx.value]
   }
 
+  // this is needed to stop an error from appearing in console when
+  // switching/refreshing the page without closing the connection
+  // TODO: move somewhere else?
+  window.onbeforeunload = function () {
+    close();
+  };
```

Comment:
I'd do it in the `onopen` listener, but despite that I think it's fine to have it - just keep this comment to keep it transparent on why it's needed.